templates/frontend/pages/index.tpl

```diff
@@ -48,7 +48,7 @@
 	{/if}
 
 	{* Announcements *}
-	{if $numAnnouncementsHomepage}
+	{if $numAnnouncementsHomepage && $announcements|count}
```

Comment:
OTOH, I think you need `$announcements|@count`. The `@` is a Smarty quirk when passing arrays into functions. From [the Smarty docs](http://www.smarty.net/docsv2/en/language.modifiers):

> If you apply a modifier to an array variable instead of a single value variable, the modifier will be applied to every value in that array. If you really want the modifier to work on an entire array as a value, you must prepend the modifier name with a @ symbol.